test/role_test.js

```diff
@@ -138,6 +138,29 @@ helper.secrets.mockSuite(helper.suiteName(__filename), ['app', 'azure'], functio
     assert(roles.some(role => role.roleId === 'thing-id:' + clientId));
   });
 
+  test('listRoleIds', async () => {
+    let result = await helper.apiClient.listRoleIds();
+    assert(result.roleIds.some(roleId => roleId === 'thing-id:' + clientId));
+  });
+
+  test('listRoleIds (limit, [continuationToken])', async () => {
+    let roleIds = [];
+    let query = {limit:1};
+    let allRoleIds = await helper.apiClient.listRoleIds();
+    
+    while (true) {
+      let result = await helper.apiClient.listRoleIds(query);
+      assume(result.roleIds.length).to.be.lessThan(2);
+      query.continuationToken = result.continuationToken;
+      roleIds = roleIds.concat(result.roleIds);
+      if (!query.continuationToken) {
+        break;
+      }
+    }
+
+    assume(roleIds).to.deeply.equal(allRoleIds.roleIds);
```

Comment:
Can you add a check here that the loop iterated more than once?

Reply:
One more thing: we do not guarantee anything about the order of roles, so it might be good to add `.sort()` to both `roleIds` and `allRoleIds.roleIds` here.